src/Components/TaskOneComponents/index.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from "react";
+import './index.css';
+
+export class Component extends React.Component<{componentName: string}>{
+    render(){
+        return(
+            <div>Hello from {this.props.componentName}</div>
+        )
+    }
+}
+
+
+export class PureComponent extends React.PureComponent{
+    render(){
+        return(
+            <div>Hello from React.PureComponent</div>
+        )
+    }
+}
+
+export const FunctionalComponent: React.FC = () => {
+    const [counter, setCounter] = useState(0)
+    
+    const handleChangeCounter = (e: React.SyntheticEvent, value: number) => {
+        setCounter(value)
+    }
+
+    return (
+        <div className="counter">
+            <div>Hello from React.FC</div>
+            <div>Current counter value is:<span className="counter__value">{counter}</span></div>
+            <div>
+                <button className="counter_button" onClick={(e) => handleChangeCounter(e, counter-1)}>-</button>
```

Comment:
There is no need to pass event to `handleChangeCounter`.

It can be just:
```js
<button className="counter_button" onClick={() => setCounter(prevCounter => prevCounter-1)}>-</button>
<button className="counter_button" onClick={() => setCounter(prevCounter => prevCounter+1)}>+</button>
```
Also if you use the previous value to update state, you should pass a function that receives the previous value and returns an updated one.

Reply:
Yeah, you right!
Pushed changes. Thank you!
Using current value from state now.